discovery_engine_core/providers/src/lib.rs

```diff
@@ -77,6 +84,16 @@ pub trait HeadlinesProvider: Send + Sync {
     async fn query_headlines(&self, query: &HeadlinesQuery<'_>) -> Result<Vec<Article>, Error>;
 }
 
+/// Abstraction over a provider for headlines only from trusted sources.
+#[async_trait]
+pub trait TrustedSourcesProvider: Send + Sync {
+    /// Query headlines.
+    async fn query_trusted_sources(
+        &self,
+        query: &TrustedSourcesQuery<'_>,
```

Comment:
i know that this is not a fat pointer per se, which i tried to express via "fat pointer in a certain sense". what i wanted to say is that the `TrustedSourcesQuery` only holds references and usizes and hence is quite similar in its structure (and to a certain extent also similar in its usage) to a fat pointer and we could therefore argue to make it `Copy` and pass it along like a fat pointer.

Reply:
No, there is no such  restriction. I mean it doesn't involve generics so no way it would hinder the "trait object" property of the provider trait.

Through saying that `TrustedSourcesQuery` is not like a fat pointer because it technically could be made copy is confusing IMHO.

Also it is not "semantically" a copy value, it just happens to currently be copy-able. But nothing says that this will stay that way in the future, as it's meant to contain all query parameters no matter weather we can copy them in or not. Furthermore we should preferable have the same interface across all queries so all queries would need to be kept copy-able.

There is no UX/codeing benefit from making it copy.

There is likely no measurable performance benefit (maybe even a hardly measurable drawback) from making it copy as it's size wise already around and maybe above the cut of where passing by reference instead of copy is faster. (it's ~64 bytes in size).

And even if it would be faster, it would only be to a degree which would be hard to measure if at all.